packages/astro/src/vite-plugin-astro-server/plugin.ts

```diff
@@ -160,7 +160,7 @@ export function createDevelopmentManifest(settings: AstroSettings): SSRManifest
 		componentMetadata: new Map(),
 		inlinedScripts: new Map(),
 		i18n: i18nManifest,
-		checkOrigin: settings.config.security?.checkOrigin ?? false,
+		checkOrigin: (settings.config.security?.checkOrigin && settings.buildOutput === "server") ?? false,
```

Comment:
This won't work in dev, because this function won't rerun on change. In dev you might go from a server build to a static one and vice versa, so this needs to get checked more often